packages/cozy-intent/src/view/components/WebviewIntentProvider.spec.tsx

```diff
@@ -59,4 +71,50 @@ describe('WebviewIntentProvider', () => {
     expect(await screen.findByText('Hello')).toBeDefined()
     expect(mockChildHandshake).not.toHaveBeenCalled()
   })
+
+  it('does not try to set cozy-bar context if cozy-bar does not exist when provided with a context', async () => {
+    global.cozy = undefined
+
+    render(
+      <WebviewIntentProvider webviewService={mockWebviewService}>
+        Hello
+      </WebviewIntentProvider>
+    )
+
+    expect(await screen.findByText('Hello')).toBeTruthy()
+    expect(mockSetWebviewContext).not.toHaveBeenCalled()
+  })
+
+  it('does not try to set cozy-bar context if cozy-bar does not exist when not provided with a context', async () => {
+    global.cozy = undefined
+
+    render(<WebviewIntentProvider>Hello</WebviewIntentProvider>)
+
+    expect(await screen.findByText('Hello')).toBeTruthy()
+    expect(mockSetWebviewContext).not.toHaveBeenCalled()
+  })
+
+  it('sets cozy-bar context if cozy-bar does exist when provided with a context', async () => {
+    global.cozy = mockCozyBar
+
+    render(
+      <WebviewIntentProvider webviewService={mockWebviewService}>
+        Hello
+      </WebviewIntentProvider>
+    )
+
+    expect(await screen.findByText('Hello')).toBeTruthy()
+    expect(mockSetWebviewContext).toHaveBeenCalledWith(mockWebviewService)
+    expect(mockSetWebviewContext).toBeCalledTimes(1)
+  })
+
+  it('sets cozy-bar context if cozy-bar does exist when not provided with a context', async () => {
+    global.cozy = mockCozyBar
+
+    render(<WebviewIntentProvider>Hello</WebviewIntentProvider>)
+
+    expect(await screen.findByText('Hello')).toBeTruthy()
```

Comment:
Please use this syntax:

it('sets cozy-bar context if cozy-bar does exist when not provided with a context', async () => {
    global.cozy = mockCozyBar

    const { findByText } = render(<WebviewIntentProvider>Hello</WebviewIntentProvider>)

    expect(await findByText('Hello')).toBeTruthy()
    expect(mockSetWebviewContext).toBeCalledWith(expect.any(WebviewService))
    expect(mockSetWebviewContext).toBeCalledTimes(1)
  })****

Reply:
Updated